### PR TITLE
feat(components/forms): Search mode in enumeration component

### PR DESCRIPTION
**Pull request by talend-sromain**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)

There is no search

**What is the new behavior?**

Search is available, but not plugged for now

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration
path for existing applications: ...

**Other information**:

### CODE REVIEW

File: output/components.sasslint.txt [hunk 1]
```diff
@@ -20,6 +20,9 @@ src/Drawer/Drawer.scss
   48:2   error    Class '.drawerStacked' should be written in lowercase with hyphens        class-name-format
   56:1   warning  Space expected between blocks                                             empty-line-between-blocks
 
+src/Enumeration/Enumeration.scss
+  19:1  warning  Space expected between blocks  empty-line-between-blocks
```
Comment (jare-talend) [inline output/components.sasslint.txt]: fix lint error

File: output/forms.eslint.txt [hunk 2]
```diff
@@ -130,7 +130,10 @@ The react/require-extension rule is deprecated. Please use the import/extensions
   151:5   error  Value must be omitted for boolean attributes  react/jsx-boolean-value
   172:5   error  Value must be omitted for boolean attributes  react/jsx-boolean-value
 
-✖ 112 problems (112 errors, 0 warnings)
+/home/travis/build/Talend/ui/packages/forms/src/widgets/EnumerationWidget/EnumerationWidget.js
+  274:3  warning  Unexpected console statement  no-console
```
Comment (jare-talend) [inline output/forms.eslint.txt]: remove console

File: packages/components/src/Enumeration/Header/HeaderSelected.component.js [hunk 3]
```diff
@@ -37,8 +37,8 @@ function HeaderSelected({ headerSelected, nbItemsSelected }) {
 
 	return (
 		<header className={headerClasses()}>
-			<span>{txtHeader}</span>
-			{nbItemsSelected > 1 && headerSelected.map((action, index) => getAction(action, index))}
+			<span><b>{txtHeader}</b></span>
```
Comment (jare-talend) [inline packages/components/src/Enumeration/Header/HeaderSelected.component.js]: avoid 2 tag span & b. Use either one or the other.
B can be avoided

File: packages/forms/src/widgets/EnumerationWidget/EnumerationWidget.js [hunk 4]
```diff
@@ -99,20 +136,20 @@ class EnumerationWidget extends React.Component {
 		// dont want to fire select item on icon click
 		event.stopPropagation();
 		const items = [...this.state.items];
-		items[value.index].displayMode = 'DISPLAY_MODE_DEFAULT';
+		items[value.index].displayMode = DISPLAY_MODE_DEFAULT;
 		items.splice(value.index, 1);
-		const countnbItems = items.filter((item) => item.isSelected).length;
+		const countnbItems = items.filter(item => item.isSelected).length;
```
Comment (jare-talend) [inline packages/forms/src/widgets/EnumerationWidget/EnumerationWidget.js]: use camelCase

File: packages/forms/src/widgets/EnumerationWidget/EnumerationWidget.js [hunk 5]
```diff
@@ -99,20 +136,20 @@ class EnumerationWidget extends React.Component {
 		// dont want to fire select item on icon click
 		event.stopPropagation();
 		const items = [...this.state.items];
-		items[value.index].displayMode = 'DISPLAY_MODE_DEFAULT';
+		items[value.index].displayMode = DISPLAY_MODE_DEFAULT;
 		items.splice(value.index, 1);
```
Comment (jare-talend) [inline packages/forms/src/widgets/EnumerationWidget/EnumerationWidget.js]: why splice?

File: packages/forms/src/widgets/EnumerationWidget/EnumerationWidget.js [hunk 6]
```diff
@@ -248,11 +272,25 @@ class EnumerationWidget extends React.Component {
...
+			headerInput: this.searchInputs,
+			displayMode: DISPLAY_MODE_SEARCH,
+		});
+
 	valueAlreadyExist(value) {
 		return this.state.items.find(item => item.values[0] === value);
 	}
 
 	updateHeaderInputDisabled(value) {
+		console.log(this);
```
Comment (jare-talend) [inline packages/forms/src/widgets/EnumerationWidget/EnumerationWidget.js]: remove console

File: packages/forms/src/widgets/EnumerationWidget/EnumerationWidget.js (hunk 5 above)
Comment (talend-sromain) [inline packages/forms/src/widgets/EnumerationWidget/EnumerationWidget.js]: Because it's some delete function, so it's removed from array

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://300.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://300.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://300.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://300.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://300.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://300.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://300.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://300.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://300.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://300.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://300.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://300.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://300.talend.surge.sh)

File: packages/components/src/Enumeration/Enumeration.scss [hunk 7]
```diff
@@ -1,13 +1,27 @@
...
+$tc-enumeration-svg-color: #666 !default;
+$tc-enumeration-font-size: 1.4rem !default;
 
 .tc-enumeration {
 	overflow: hidden;
 	display: flex;
 	flex-direction: column;
 	height: $tc-enumeration-height;
 
+	:global(input[type=text]) {
```
Comment (jlame) [inline packages/components/src/Enumeration/Enumeration.scss]: is the global necessary ?

Comment (jlame) [inline packages/components/src/Enumeration/Enumeration.scss]: same goes for other btn btn-link

File: packages/components/src/Enumeration/Items/Item/Item.component.js [hunk 8]
```diff
@@ -58,15 +58,86 @@ function Item({ id, item }) {
...
+						{index === matchIndex.length + 1 ? null :
+							label.substring(matchIndex + searchCriteria.length, matchIndexes[index + 1])
+						}
+					</span>
+				))
+			}
+		</span>);
+	}
+
+	let actionLabel = (<Action
```
Comment (jlame) [inline packages/components/src/Enumeration/Items/Item/Item.component.js]: what about taking this out in a function ?

**Comment by jlame** [comment]
we could have move some function to make them static but this force us to pass a lot of args which is not really readable.

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://300.talend.surge.sh)

**Pull request merged**